### PR TITLE
Feature/geode 6762 2

### PR DESCRIPTION
**Pull request by mivanac**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [*] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [*] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [*] Is your initial contribution a single, squashed commit?

- [*] Does `gradlew build` run cleanly?

- [*] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by mivanac** [comment]
Are there any comments for this PR?

**Comment by kirklund** [comment]
@mivanac Thanks again for another contribution! I'll email the dev list for you to try and encourage some reviews. Please feel free to email the dev list anytime about your PRs.

You could probably tidy up the PR title and the commit messages to encourage more reviews as well. It's kind of hard to know what this PR pertains to without studying the Jira ticket GEODE-6762. This in no way justifies no reviews though!

For git commit messages, we try to follow [Chris Beams' How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/). For example, you might squash your commits to: "GEODE-6762: Prevent unnecessary PartitionedRegion callback events". That would probably make a good PR title as well.

File: geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/CacheClientNotifier.java [hunk 1]
```diff
@@ -608,7 +608,7 @@ public static boolean singletonHasClientProxies() {
   }
 
   private boolean hasClientProxies() {
-    return !this._initClientProxies.isEmpty() || !this._clientProxies.isEmpty();
+    return !this._clientProxies.isEmpty() || !this._initClientProxies.isEmpty();
```
Comment (dschneider-pivotal) [inline geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/CacheClientNotifier.java]: 0. initial condition: _clientProxies.isEmpty() and !_initClientProxies.isEmpty()
1. Thread 1 calls _clientProxies.isEmpty() which returns true
2. Thread 2 calls addClientProxy so now: !_clientProxies.isEmpty() and _initClientProxies.isEmpty()
3. Thread 1 calls _initClientProxies.isEmpty() which returns true
So Thread 1 returns false from hasClientProxies even though we have one client proxy.
The fix is to check _initClientProxies first and then _clientProxies

Comment (mivanac) [reply geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/CacheClientNotifier.java]: Sorry, but when resolving merge conflicts, I deleted wrong part.

**Comment by dschneider-pivotal** [comment]
The integration test that failed looks like: https://issues.apache.org/jira/browse/GEODE-526.
But it was fixed and closed a long time ago. See if a rerun on you branch passes and if reopen GEODE-526 and point it at: https://concourse.apachegeode-ci.info/builds/79624

**Pull request merged**